web-extensions/chrome/engine/bookmycomics.js

```diff
@@ -5,24 +5,43 @@
  *
  * @class BmcEngine
  */
-function BmcEngine() {
+function BmcEngine(comicName, chapter, page) {
     this._db = new BmcDataAPI();
     console.log('Instanciated BmcDataAPI');
-    this._ui = new BmcUI();
+    this._messaging = new BmcMessagingHandler();
+    console.log('Instanciated BmcMEssagingHandler');
+    this._ui = new BmcUI(this._messaging);
     console.log('Instanciated BmcUI');
 
     // Re-use DOM-style events by using a null-text node underneath
     this._eventCore = document.createTextNode(null);
     this.addEventListener = this._eventCore.addEventListener.bind(this._eventCore);
     this.removeEventListener = this._eventCore.removeEventListener.bind(this._eventCore);
     this.dispatchEvent = this._eventCore.dispatchEvent.bind(this._eventCore);
+
+    // A bit of stateful data, so that we can avoid re-checking the storage
+    // everytime (and thus speed-up a bit the logic that spawn the UI bits)
+    this._comic = {
+        name: comicName,
+        chapter,
+        page,
+        id: undefined,
+	memoizing: false,
```

Comment:
Failed indent (please use whitespaces instead of tabs ^^, however, if you prefer using tabs we can discuss about it).

Reply:
Nah, I just missed those. Will fix